PokemonGo/RocketAPI/Window/PokeUi.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using PokemonGo.RocketAPI.Enums;
+using PokemonGo.RocketAPI.GeneratedCode;
+
+namespace PokemonGo.RocketAPI.Window
+{
+    public partial class PokeUi : Form
+    {
+        public PokeUi()
+        {
+            InitializeComponent();
+            ClientSettings = Settings.Instance;
+        }
+
+        private void PokeUi_Load(object sender, EventArgs e)
+        {
+            Execute();
+        }
+
+        public static ISettings ClientSettings;
+
+        private async void Execute()
+        {
+            button1.Enabled = false;
+
+            var client = new Client(ClientSettings);
+
+            try
+            {
+                switch (ClientSettings.AuthType)
+                {
+                    case AuthType.Ptc:
+                        await client.DoPtcLogin(ClientSettings.PtcUsername, ClientSettings.PtcPassword);
+                        break;
+                    case AuthType.Google:
+                        await client.DoGoogleLogin();
+                        break;
+                }
+                //
+                await client.SetServer();
+                var inventory = await client.GetInventory();
+                var pokemons =
+                    inventory.InventoryDelta.InventoryItems
+                    .Select(i => i.InventoryItemData?.Pokemon)
+                        .Where(p => p != null && p?.PokemonId > 0)
+                        .OrderByDescending(key => key.Cp);
+                var families = inventory.InventoryDelta.InventoryItems
+                    .Select(i => i.InventoryItemData?.PokemonFamily)
+                    .Where(p => p != null && (int) p?.FamilyId > 0)
+                    .OrderByDescending(p => (int)p.FamilyId);
+                    
+                    
+                
+                
+                var imageList = new ImageList { ImageSize = new Size(50, 50) };
+                listView1.ShowItemToolTips = true;
+
+                foreach (var pokemon in pokemons)
+                {
+
+                    var pokemonImage = GetPokemonImage((int)pokemon.PokemonId);
```

Comment:
This is going to download the images from every time you open the window.   You should really consider caching them both in memory and on disk.